platform/windows/detect.py

```diff
@@ -446,12 +473,29 @@ def configure_mingw(env):
         ]
     )
 
-    env.Append(CPPDEFINES=["VULKAN_ENABLED"])
-    if not env["use_volk"]:
-        env.Append(LIBS=["vulkan"])
+    if env["vulkan"]:
+        env.Append(CPPDEFINES=["VULKAN_ENABLED"])
+        if not env["use_volk"]:
+            env.Append(LIBS=["vulkan"])
+
+    if env["d3d12"]:
+        env.AppendUnique(CPPDEFINES=["D3D12_ENABLED"])
+        env.Append(LIBS=["d3d12", "dxgi", "dxguid"])
+
+        # DXC
+        # (Overriding those in the Windows SDK.)
+        env.Prepend(CPPPATH=[env["DXC_PATH"] + "/inc"])
+        env.Prepend(LIBPATH=[env["DXC_PATH"] + "/lib/x64"])
```

Comment:
I assume there's no 32-bit version (at least it's not part of the precompiled DXC releases)?

For Windows on ARM support, this (and `dll` copy code) need to be changed to `/lib/arm64`, the rest of the DX renderer code seems to building fine (tested cross building from x86-64 Windows using VS 2022).

<details>
 <summary>Running on ARM Windows VM</summary>

<img width="1988" alt="msvc_dx12" src="https://user-images.githubusercontent.com/7645683/184532907-889a7a4f-f726-4ed5-85fa-460a693b232e.png">

</details>

Reply:
I really appreciate you tested this!

Build scripts fixed.